src/uct/sm/scopy/cma/cma_ep.c

```diff
@@ -58,43 +58,32 @@ UCS_CLASS_DEFINE(uct_cma_ep_t, uct_scopy_ep_t)
 UCS_CLASS_DEFINE_NEW_FUNC(uct_cma_ep_t, uct_ep_t, const uct_ep_params_t *);
 UCS_CLASS_DEFINE_DELETE_FUNC(uct_cma_ep_t, uct_ep_t);
 
-static UCS_F_NOINLINE void
-uct_cma_ep_tx_error(uct_cma_ep_t *ep, const char *cma_call_name,
-                    ssize_t cma_call_ret, int cma_call_errno,
-                    const struct iovec *local_iov, size_t local_iov_cnt,
-                    const struct iovec *remote_iov)
+void uct_cma_ep_tx_error(uct_cma_ep_t *ep, uct_scopy_tx_op_t tx_op,
+                         ssize_t ret, int op_errno, struct iovec *local_iov,
+                         size_t local_iov_cnt, struct iovec *remote_iov)
 {
-    uct_base_iface_t *iface = ucs_derived_of(ep->super.super.super.iface,
-                                             uct_base_iface_t);
-    UCS_STRING_BUFFER_ONSTACK(local_iov_str, 256);
-    UCS_STRING_BUFFER_ONSTACK(remote_iov_str, 256);
-    ucs_log_level_t log_lvl;
-    ucs_status_t status;
-
-    status  = uct_iface_handle_ep_err(&iface->super, &ep->super.super.super,
-                                      UCS_ERR_CONNECTION_RESET);
-    log_lvl = uct_base_iface_failure_log_level(iface, status,
-                                               UCS_ERR_CONNECTION_RESET);
-
-    /* Dump IO vector */
-    ucs_string_buffer_append_iovec(&local_iov_str, local_iov, local_iov_cnt);
-    ucs_string_buffer_append_iovec(&remote_iov_str, remote_iov, 1);
-
-    ucs_log(log_lvl, "%s(pid=%d {%s}-->{%s}) returned %zd: %s", cma_call_name,
-            ep->remote_pid, ucs_string_buffer_cstr(&local_iov_str),
-            ucs_string_buffer_cstr(&remote_iov_str), cma_call_ret,
-            strerror(cma_call_errno));
+    UCS_STRING_BUFFER_ONSTACK(remote_pid_strb, 32);
+    UCS_STRING_BUFFER_ONSTACK(ret_strb, 32);
+
+    ucs_string_buffer_appendf(&remote_pid_strb, "remote_pid=%d",
+                              ep->remote_pid);
+    ucs_string_buffer_appendf(&ret_strb, "ret=%zd", ret);
+
+    uct_scopy_ep_tx_error(
+            &ep->super, &remote_pid_strb, uct_cma_ep_fn[tx_op].name, &ret_strb,
+            op_errno, sizeof(struct iovec), (const void*)local_iov,
+            local_iov_cnt, (const void*)remote_iov,
```

Comment:
maybe can remove the cast to const void*

Reply:
removed